libs/vm/vm.py

```diff
@@ -58,6 +62,16 @@ def _filter_out_none_values(data: list[tuple[str, Any]]) -> dict[str, Any]:
     def login_params(self) -> dict[str, str]:
         return py_config["os_login_param"][self._os_distribution]
 
+    @retry(
+        wait_timeout=TIMEOUT_9MIN,
```

Comment:
1.
I think that 9 minutes is a lot.
Don't forget that we check the console after the VM completes its start up. At this point, if something is eventually broken with the console then we should fail earlier, after a reasonable grace. 
2 minutes sound enough to me.

2.
Are you sure this is the correct place to implement the retry? We have the issue only in the console login, so I think that retrying here is out of scope.

3.
I think that a time-based retry is not right here, and that a iterations-based retry makes more sense (i.e. try connecting 3 times, and if all of them failed - declare failure).

4.
I don't think it's necessary to apply the retry both here and in `_connect` (later on in this PR). Based on my comment 2, I think that `_connect` makes more sense.